hazelcast/src/main/java/com/hazelcast/internal/networking/nio/NioOutboundPipeline.java

```diff
@@ -51,17 +52,26 @@
         extends NioPipeline
         implements Supplier<OutboundFrame>, OutboundPipeline {
 
+    private enum State {
+        UNSCHEDULED,
```

Comment:
I feel like a description for each state and the transitions would be very helpful in reviewing this PR and future maintenance.

Reply:
Done